### PR TITLE
Update to trivia

### PR DESCRIPTION
**Pull request by fraserbaigent**
After cocking around with git for an hour, I have cleaned up some of the module ( a little) and added functionality so the same question won't be asked in 200 times. 

It might be better to just return until the next execution rather than using a while loop to check if the question is unique, that way the bot will never exceed the api request limit. Also added a streak message.

### CODE REVIEW

File: pajbot/modules/trivia.py [hunk 1]
```diff
@@ -77,77 +75,129 @@ def __init__(self):
...
             self.question['answer'] = self.question['answer'].replace('a ', '')
+            
         elif self.question['answer'].startswith('an '):
             self.question['answer'] = self.question['answer'].replace('an ', '')
+            
         if self.question['answer'].lower().startswith('the '):
             self.question['answer'] = self.question['answer'].replace('the ', '')
-
+                    
+    ############################################################################
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: This isn't needed

File: pajbot/modules/trivia.py [hunk 2]
```diff
@@ -77,77 +75,129 @@ def __init__(self):
...
-                    self.question['category'] = base64.b64decode(resjson['category']).decode('utf-8')
-                else:
+                    category = random.choice([9, 11, 15, 17, 18, 20, 21, 22, 23,
+                                              24, 26, 27, 29, 30])
+                    while not new_question:                                       
+                        r = requests.get('https://opentdb.com/api.php?amount=1&category={}&type=multiple&encode=base64'.format(category))
+                        try:
+                            resjson = r.json()['results'][0]
+                        except:
+                            return
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: Since you're putting it in a while loop you can use continue

File: pajbot/modules/trivia.py [hunk 3]
```diff
@@ -77,77 +75,129 @@ def __init__(self):
...
+                        if self.question['question'] not in self.recent_questions:
+                            self.recent_questions.append(self.question['question'])
+                            self.question['category'] = self.question['category'].replace('_', ' ')
+                            new_question = True
+                            
+            # Remove oldest question
+            if len(self.recent_questions) > self.q_memory:
+                del self.recent_questions[0]
+                
+            if not self.question['answer'] or not self.question['question'] or \
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: Make a list of bad phrases, 
`if any(badPhrase in self.question['answer'] for badPhrase in badPhrases):  go back`

File: pajbot/modules/trivia.py [hunk 4]
```diff
@@ -163,7 +213,7 @@ def step_announce(self):
             if self.jservice:
                 self.bot.safe_me('PogChamp A new question has begun! In the category "{0[category][title]}", the question/hint/clue is "{0[question]}" Bruh'.format(self.question))
             else:
-                self.bot.safe_me('PogChamp A new question has begun! In the category "{0[category]}", the question is: "{0[question]}"'.format(self.question))
+                self.bot.safe_me('PogChamp A new question has begun! The question is: "{0[question]}"'.format(self.question))
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: Category should be fine to stay

File: pajbot/modules/trivia.py [hunk 5]
```diff
@@ -5,9 +5,7 @@
 
 import Levenshtein
 import requests
-import random
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: randint is used

File: pajbot/modules/trivia.py [hunk 6]
```diff
@@ -77,77 +75,129 @@ def __init__(self):
         self.step = 0
         self.last_step = None
         self.correct_dict = {}
-        self.gazCategories = ['W_OMEGALUL_W', 'Vietnam', 'Video_Games', 'Video Games', 'Twitch', 'Sports', 'Spongebob', 'Jokes', 'Science', 'Programming', 'Music',
-                              'Memes', 'Math', 'Maths', 'Movies', 'Languages', 'History', 'Geography', 'Gachimuchi', 'Gachi', 'Emotes', 'Bees', 'Country', 'Books']
+        
+        self.gazCategories = ['W_OMEGALUL_W', 'Vietnam', 'Video_Games',
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: I haven't gone over them but please make sure that you haven't accidentally removed a category

File: pajbot/modules/trivia.py (hunk 3 above)
Comment (fraserbaigent) [reply pajbot/modules/trivia.py]: I've done this and rolled it in with the check on if the question is in the recent questions .

File: pajbot/modules/trivia.py (hunk 5 above)
Comment (fraserbaigent) [reply pajbot/modules/trivia.py]: no idea why this was removed

File: pajbot/modules/trivia.py (hunk 6 above)
Comment (fraserbaigent) [reply pajbot/modules/trivia.py]: all still there

File: pajbot/modules/trivia.py [hunk 7]
```diff
@@ -77,77 +76,130 @@ def __init__(self):
...
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: Needs some additions:

['Vietnam', 'Video Games', 'Sports', 'Jokes', 'Programming', 'Memes', 'Maths', 'Languages', 'HTTP', 'Gachimuchi', 'Emotes', 'Country', 'D DansGame TA', 'AdmiralBulldog']

Comment (fraserbaigent) [reply pajbot/modules/trivia.py]: Added the ones that were missing but left jokes out, it's always long and convoluted

File: pajbot/modules/trivia.py [hunk 8]
```diff
@@ -77,77 +76,101 @@ def __init__(self):
...
-            if (len(self.question['answer']) == 0 or len(self.question['question']) <= 1 or 'href=' in self.question['answer']
-                or 'Which of these' in self.question['question'] or 'Which one of these' in self.question['question']
-                or 'Which of the following' in self.question['question']):
-
-                self.question = None
-                return
+                    # Load from gazatu and RTD
+                    chosenInt = random.randint(0, 10)
+                    if chosenInt <= 5:
+                        while not new_question:
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: Double new_question

File: pajbot/modules/trivia.py [hunk 9]
```diff
@@ -331,11 +366,11 @@ def load_commands(self, **options):
                         can_execute_with_whisper=True,
                         ),
                     'skip': pajbot.models.command.Command.raw_command(
-                        self.command_skip,
-                        level=500,
-                        delay_all=0,
-                        delay_user=0,
-                        can_execute_with_whisper=True,
+	                self.command_skip,
```
Comment (DatGuy1) [inline pajbot/modules/trivia.py]: Did you change spaces to tabs/vice versa? Just wondering

Comment (fraserbaigent) [reply pajbot/modules/trivia.py]: I just deleted a line by accident, got some weird config on emacs so it replaces tabs with spaces so things will always work.

**Pull request merged**